### PR TITLE
[multi_ptr] Add subscript operator to multi_ptr.

### PR DESCRIPTION
**Pull request by Naghasan**
Add subscript operator to make multi_ptr more a pointer-like object.

### CODE REVIEW

**Comment by keryell** [comment]
`std::ptrdiff_t` looks like a good choice.

File: adoc/chapters/programming_interface.adoc [hunk 1]
```diff
@@ -9074,6 +9074,16 @@ template<access::address_space ASP, access::decorated IsDecorated>
 
 Move the value of the left hand side [code]#multi_ptr# into the [code]#generic_ptr#.
 
+a@
+[source]
+----
+template <typename value_type, access::address_space Space, access::decorated DecorateAddress>
+    reference operator[](std::ptrdiff_t i) const
```
Comment (gmlueck) [inline adoc/chapters/programming_interface.adoc:9081]: Why is the operator shown as a template function here?  It's not a template in the code synopsis.

Comment (Naghasan) [reply adoc/chapters/programming_interface.adoc:9081]: For consistency with other declarations.

I do agree however the spec needs a clean up on that point, it doesn't quite make sense

Comment (gmlueck) [reply adoc/chapters/programming_interface.adoc:9081]: I see.  Yes, all the `multi_ptr` operators should be cleaned up.  It's probably better to do that in a separate PR.

Comment (keryell) [reply adoc/chapters/programming_interface.adoc:9081]: Yes, but at least remove the template here so we can merge this first.

Comment (Naghasan) [reply adoc/chapters/programming_interface.adoc]: sure, done

Comment (Naghasan) [reply adoc/chapters/programming_interface.adoc]: > Yes, all the multi_ptr operators should be cleaned up. It's probably better to do that in a separate PR.

Opened one for this https://github.com/KhronosGroup/SYCL-Docs/pull/154

File: adoc/chapters/what_changed.adoc [hunk 2]
```diff
@@ -359,6 +359,7 @@ Changes in [code]#multi_ptr# interface:
     underlying pointer as an unannotated pointer;
   * addition of the member function [code]#get_decorated# which returns the
     underlying pointer as an annotated pointer.
```
Comment (keryell) [inline adoc/chapters/what_changed.adoc:361]: Since now there is a new line after:
```suggestion
    underlying pointer as an annotated pointer;
```

**Review by keryell — CHANGES_REQUESTED** [review]
Almost there!

File: adoc/chapters/what_changed.adoc (hunk 2 above)
Comment (Naghasan) [reply adoc/chapters/what_changed.adoc]: Indeed, suggestion committed.

**Review by keryell — APPROVED** [review]
Thanks!

**Comment by keryell** [comment]
@Naghasan ping.

**Comment by Naghasan** [comment]
Sorry I missed the previous message, conflict resolved

**Review by keryell — APPROVED** [review]
Thanks.

**Pull request merged**